src/remarx/candidate_mentions.py

```diff
@@ -0,0 +1,199 @@
+    doc = pipeline(text) if isinstance(text, str) else text
+
+    return " ".join(
+        [lemmatize_sentence(s, drop_punct=drop_punct) for s in doc.sentences]
+    )
+
+
+def find_sentences_with_phrase(
+    search_phrase: str,
+    text: str | Document,
+    pipeline: None | Pipeline = None,
+) -> Generator[Sentence, None, None]:
+    """
+    Returns a generator of sentences within a given text that contains the search phrase
+    ignoring inflection and capitalization.
+    """
+    # Intialize pipeline if one not given
+    if pipeline is None:
+        pipeline = Pipeline(
+            lang=LANG, processors=PROCS, download_method=DownloadMethod.REUSE_RESOURCES
+        )
+    # Lemmatize search phrase
+    lem_phrase = lemmatize_text(search_phrase, pipeline=pipeline).lower()
+
+    # Construct stanza document if needed
+    doc = pipeline(text) if isinstance(text, str) else text
+
+    # Lemmatize & check if a sentence contains a search phrase
+    for sentence in doc.sentences:
+        lem_sent = lemmatize_sentence(sentence).lower()
+        if " " + lem_phrase + " " in " " + lem_sent + " ":
+            yield sentence
+
+
+def save_candidate_sentences(
+    input_texts: Iterable[pathlib.Path],
+    search_phrases: list[str],
+    output_csv: pathlib.Path,
+    case_sensitive: bool = False,
+    delimiter: str = " | ",
+) -> None:
+    """
+    Finds sentences within the input texts that contain the search phrases and
+    saves the results as a CSV file with the following fields:
+
+    * file: name of text file
+    * sent_idx: sentence-level index within text file
+    * char_idx: character-level index within text file
+    * sentence: sentence text
+    * phrases: list of matching phrases
+    """
+    # Initialize stanza pipeline
+    pipeline = Pipeline(
+        lang=LANG, processors=PROCS, download_method=DownloadMethod.REUSE_RESOURCES
+    )
+
+    # Lemmatize & lowercase pharses (for efficiency)
+    lem_phrases = [lemmatize_text(p, pipeline=pipeline).lower() for p in search_phrases]
+
+    fieldnames = ["file", "sent_idx", "char_idx", "sentence", "phrases"]
+    with open(output_csv, mode="w", encoding="utf-8-sig", newline="") as f:
+        writer = csv.DictWriter(f, fieldnames=fieldnames)
+        writer.writeheader()
+
+        file_progress = tqdm(input_texts)
+        for text_file in file_progress:
+            file_progress.set_description(f"Processing {text_file.stem}")
+            doc = pipeline(text_file.read_text())
+            for i, sent in enumerate(doc.sentences):
+                # Lemmatize & lowercase sentence
+                lem_sent = lemmatize_sentence(sent).lower()
+                # Determine which phrases the sentence contains
+                matches = []
+                for j, phrase in enumerate(lem_phrases):
+                    # Add spacing to ensure word-level boundaries
+                    if f" {phrase} " in f" {lem_sent} ":
+                        matches.append(search_phrases[j])
```

Comment:
I see now from looking at the output that I wasn't understanding your comment about aggregation - of course sentences that match the more specific variant will also match the less specific case. If we knew the more specific variant you could privilege that one first and not bother checking others, but I like the format you came up with for the output (and the extra check is pretty trivial).